File: backend/src/main/java/com/viniciusfinger/rsflood/model/Necessity.java
package com.viniciusfinger.rsflood.model;

import com.viniciusfinger.rsflood.enums.NecessityCategory;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Necessity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String description;

    private String quantity;

    @Enumerated(EnumType.STRING)
    private NecessityCategory category;

}

File: frontend/src/app/model/necessity.ts
import { NecessityCategory } from "./enum/necessity-category";

export class Necessity {
    id: number | null;
    description: string;
    quantity: number;
    category: NecessityCategory;

    constructor(id: number | null, description: string, quantity: number, category: NecessityCategory) {
        this.id = id;
        this.description = description;
        this.quantity = quantity;
        this.category = category;
    }
}
